python/cudf/cudf/core/udf/utils.py

```diff
@@ -356,14 +376,20 @@ def _post_process_output_col(col, retty):
     return as_column(col, retty)
 
 
+# The only supported data layout in NVVM.
+# See: https://docs.nvidia.com/cuda/nvvm-ir-spec/index.html?#data-layout
+_nvvm_data_layout = (
+    "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-"
+    "i128:128:128-f32:32:32-f64:64:64-v16:16:16-v32:32:32-"
+    "v64:64:64-v128:128:128-n16:32:64"
+)
```

Comment:
Why does this need to be hardcoded now? Would the old logic using `nvvm.data_layout` not have picked it up automatically?

Reply:
There's only one possible datalayout for CUDA, and the place where Numba keeps the datalayout is an implementation detail, so to avoid this breaking underneath us again we should really just keep our own copy of it.